### PR TITLE
Remove six as a dependency (closes #425)

### LINKED ISSUE #409: Drop support for Python 2.7
_“DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won’t be maintained after that date. A future version of pip will drop support for Python 2.7.“_

The Tech Arch team has approved dropping Python 2.7 support.

Please note that `dcplib`, on which dcp-cli depends, is dropping support for Python 2.7 after version 2.1.2.

### LINKED ISSUE #425: Make `--layout=bundle` the default
Originally brought up by @ESapenaVentura (on Slack, `#data-store`):

> I tried hca dss download-manifest  this morning with a modified manifest (deleted almost all lines because I just wanted to test how it function), and it downloaded the files to the temporary hidden folder and, in the end, it modified the manifest to indicate the hidden folders where the files (with modified names) are. I tried downloading by bundle id and the behaviour is completely different, downloads the bundle at a non-hidden directory with the original name of the file. Is the first behaviour intended?

### LINKED ISSUE #435: Delete "six" as a dependency.
We no longer support python 2.7, so this should be removed.  Depends on #430.

### PR DESCRIPTION
**Pull request by natanlao**
Closes #409 
Closes #435 
Related to #430

### CODE REVIEW

**Comment by natanlao** [comment]
@jessebrennan Ack, good catch, thanks!

File: hca/upload/upload_area_uri.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-from six.moves.urllib.parse import urlparse
+# -*- coding: utf-8 -*-
```
Comment (jessebrennan) [inline hca/upload/upload_area_uri.py]: Also, what's up with this like?

**Comment by natanlao** [comment]
> Nice. When you merge, can you consolidate 60aa8f4 and 291f26c? (The other commit makes sense to keep separate.)

Yep! I usually wait until review has completed to squash so that it doesn't interfere with comments.

**Comment by jessebrennan** [comment]
That's an excellent strategy.

**Pull request merged**